std/array.d

```diff
@@ -2840,6 +2840,20 @@ if (isDynamicArray!A)
         }
     }
 
+  // emplace an item onto an array.
+  void emplacePut(Args...)(Args args) {
+	import std.conv: emplaceRef;
```

Comment:
I will be frank, emplacePut is a silly sounding name. I wanted to do emplace, because appender.emplace(...) is unambiguous, since an appender can only emplace by appending, but it might be confusing for other definitions of emplace, that don't append anything. I wanted to overload appender.put(...) for when appending to a non-copyable type, but having totally different semantics for put(...) depending on what's in there is confusing. So emplacePut is the best I could come up with.